reV/handlers/multi_year.py

```diff
@@ -126,15 +128,53 @@ def _copy_dset(self, source_h5, dset, meta=None, pass_through=False):
             self._create_dset(dset_out, ds_data.shape, ds_dtype,
                               chunks=ds_chunks, attrs=ds_attrs, data=ds_data)
 
+    @staticmethod
+    def parse_source_files_pattern(source_files):
+        """Parse a source_files pattern that can be either an explicit list of
+        source files or a unix-style /filepath/pattern*.h5 and either way
+        return a list of explicit filepaths.
+
+        Parameters
+        ----------
+        source_files : list | str
+            List of .h5 files to collect datasets from. This can also be a
+            unix-style /filepath/pattern*.h5 to find .h5 files to collect,
+            however all resulting files must be .h5 otherwise an exception will
+            be raised. NOTE: .h5 file names much indicate the year the data
+            pertains to
+
+        Returns
+        -------
+        source_files : list
+            List of .h5 filepaths.
+        """
+
+        if isinstance(source_files, (list, tuple)):
+            pass
+        elif isinstance(source_files, str) and '*' in source_files:
+            source_files = glob.glob(source_files)
+        elif isinstance(source_files, str):
+            source_files = [source_files]
+
+        if not all(fp.endswith('.h5') for fp in source_files):
+            msg = ('Non-h5 files cannot be collected: {}'.format(source_files))
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        return source_files
+
     def collect(self, source_files, dset, profiles=False, pass_through=False):
         """
         Collect dataset dset from given list of h5 files
 
         Parameters
         ----------
-        source_files : list
-            List of .h5 files to collect datasets from
-            NOTE: .h5 file names much indicate the year the data pertains to
+        source_files : list | str
+            List of .h5 files to collect datasets from. This can also be a
+            unix-style /filepath/pattern*.h5 to find .h5 files to collect,
+            however all resulting files must be .h5 otherwise an exception will
+            be raised. NOTE: .h5 file names much indicate the year the data
```

Comment:
```suggestion
            be raised. NOTE: .h5 file names must indicate the year the data
```

Reply:
Good catch! adding in a general PR review edits commit